### PR TITLE
new recipe: elforkane

### PR DESCRIPTION
**Pull request by mazbrili**
elforkane Qt quran browser application

### CODE REVIEW

File: app-text/elforkane/elforkane-1.2.recipe [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+PROVIDES="
+	elforkane$secondaryArchSuffix = $portVersion
+	app:Elforkane = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	ayat_tarajem_data
+	ayat_tafasir_data
+	lib:libGL$secondaryArchSuffix
```
Comment (extrowerk) [inline app-text/elforkane/elforkane-1.2.recipe]: Are you sure about it?

File: app-text/elforkane/elforkane-1.2.recipe [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+{
+	qmake
+	make $jobArgs
+}
+
+INSTALL()
+{
+    mkdir -p $appsDir/Elforkane/{bin,share}
+
+    cp -rf usr/bin/elforkane $appsDir/Elforkane/bin/Elforkane
```
Comment (extrowerk) [inline app-text/elforkane/elforkane-1.2.recipe]: Do not indent with spaces.

File: app-text/elforkane/elforkane-1.2.recipe [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+	rm -rf usr/share/elforkane/translat
+	rm -rf usr/share/elforkane/tafasir
+    cp -rf usr/share/* $appsDir/Elforkane/share
```
Comment (extrowerk) [inline app-text/elforkane/elforkane-1.2.recipe]: Also here

File: app-text/elforkane/elforkane-1.2.recipe [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+	devel:libQt5Xml$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:gcc$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	cmd:qmake$secondaryArchSuffix >= 5
+	cmd:unzip
```
Comment (extrowerk) [inline app-text/elforkane/elforkane-1.2.recipe]: Is it required?

File: app-text/elforkane/elforkane-1.2.recipe [hunk 5]
```diff
@@ -0,0 +1,86 @@
+SUMMARY="The electronic Holy Quran browser"
+DESCRIPTION="Electronic Holy Quran browser and reciter. \
+"
+COPYRIGHT="1999-2019 Abou Zakaria"
+HOMEPAGE="https://github.com/zakariakov/elforkane"
+LICENSE="GNU GPL v3"
+REVISION="1"
+SOURCE_URI="https://github.com/zakariakov/elforkane/archive/$portVersion.tar.gz"
+CHECKSUM_SHA256="25f5352472e99e6dd018f6003408a793ee85f140a381d58b7de463e9b0bc2003"
+
```
Comment (extrowerk) [inline app-text/elforkane/elforkane-1.2.recipe]: No need for this empty line.

File: app-text/elforkane/elforkane-1.2.recipe [hunk 6]
```diff
@@ -0,0 +1,86 @@
+SUMMARY="The electronic Holy Quran browser"
+DESCRIPTION="Electronic Holy Quran browser and reciter. \
+"
```
Comment (extrowerk) [inline app-text/elforkane/elforkane-1.2.recipe]: This can go in previous line.

**Comment by extrowerk** [comment]
Also no need for icons here and main binary shoul be in $appsDir, not in bin.

**Comment by extrowerk** [comment]
Requires/Build requires are incomplete/incorrect, here is a list about the rependencies:
```
  NEEDED               libQt5MultimediaWidgets.so.5
  NEEDED               libQt5Multimedia.so.5
  NEEDED               libQt5PrintSupport.so.5
  NEEDED               libQt5Widgets.so.5
  NEEDED               libQt5Gui.so.5
  NEEDED               libQt5Sql.so.5
  NEEDED               libQt5Network.so.5
  NEEDED               libQt5Core.so.5
  NEEDED               libGL.so.1
```
So libGL is really needed here, but you should add the others missing ones too.

File: app-text/elforkane/elforkane-1.2.recipe (hunk 5 above)
Comment (mazbrili) [reply app-text/elforkane/elforkane-1.2.recipe]: ok done

File: app-text/elforkane/elforkane-1.2.recipe (hunk 6 above)
Comment (mazbrili) [reply app-text/elforkane/elforkane-1.2.recipe]: ok done

File: app-text/elforkane/elforkane-1.2.recipe (hunk 3 above)
Comment (mazbrili) [reply app-text/elforkane/elforkane-1.2.recipe]: ok done

File: app-text/elforkane/elforkane-1.2.recipe (hunk 2 above)
Comment (mazbrili) [reply app-text/elforkane/elforkane-1.2.recipe]: ok done

File: app-text/elforkane/elforkane-1.2.recipe (hunk 1 above)
Comment (mazbrili) [reply app-text/elforkane/elforkane-1.2.recipe]: yes it use libgl

**Comment by mazbrili** [comment]
> Also no need for icons here and main binary shoul be in $appsDir, not in bin.

so .. it need patch to original source code.. i'll try find time to check it further

**Comment by mazbrili** [comment]
please help to check again @extrowerk

File: app-text/elforkane/elforkane-1.2.recipe (hunk 4 above)
Comment (mazbrili) [reply app-text/elforkane/elforkane-1.2.recipe]: ..after recheck it is not.. removed.. thanks @extrowerk

**Comment by extrowerk** [comment]
```
Collecting files to be packaged ...
cp: cannot create regular file '/packages/elforkane-1.2-1/.self/apps/Elforkane/bin/Elforkane': No such file or directory
Warning: Command '['bash', '-c', '. /wrapper-script']' returned non-zero exit status 1.
Error: Build has failed - stopping.
```

**Comment by extrowerk** [comment]
bin directory not exists.
You added the attributes to a non-existing file
The program starts, but doesn't seems to find itsh data, at least no text shows up at all. Sadly i have no idea how  to use this program, maybe i misunderstood something.

File: app-text/elforkane/elforkane-1.2.recipe [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+LICENSE="GNU GPL v3"
+REVISION="1"
+SOURCE_URI="https://github.com/zakariakov/elforkane/archive/$portVersion.tar.gz"
+CHECKSUM_SHA256="25f5352472e99e6dd018f6003408a793ee85f140a381d58b7de463e9b0bc2003"
+SOURCE_FILENAME="elforkane-$portVersion.tar.gz"
+SOURCE_DIR="elforkane-$portVersion"
+ADDITIONAL_FILES="elforkane.rdef.in"
+PATCHES="elforkane-$portVersion.patchset"
+
+ARCHITECTURES="?x86_gcc2 x86 x86_64"
```
Comment (Begasus) [inline app-text/elforkane/elforkane-1.2.recipe]: You need to set x86_gcc2 as broken

File: app-text/elforkane/elforkane-1.2.recipe [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+
+BUILD()
+{
+	qmake
+	make $jobArgs
+}
+
+INSTALL()
+{
+	mkdir -p $appsDir/Elforkane/lib
```
Comment (Begasus) [inline app-text/elforkane/elforkane-1.2.recipe]: This should fix the install part:
```diff
- 	mkdir -p $appsDir/Elforkane/lib
+ 	mkdir -p $appsDir/Elforkane/{bin,lib}
```

**Comment by Begasus** [comment]
elforkane keeps sitting in the systray after launch, no way to close it there aside from killing it with PC, also, after installing the data packages they still don't show up in the menu (don't have a clue also how the app works though) :)

**Comment by Begasus** [comment]
@mazbrili any update on this?

**Comment by mazbrili** [comment]
> @mazbrili any update on this?

i want to check it this week.. hope problem can be fix

File: app-text/elforkane/elforkane-1.2.recipe (hunk 7 above)
Comment (mazbrili) [reply app-text/elforkane/elforkane-1.2.recipe]: ok. marked as broken

File: app-text/elforkane/elforkane-1.2.recipe (hunk 8 above)
Comment (mazbrili) [reply app-text/elforkane/elforkane-1.2.recipe]: ...its come  from wrong copy location.. fixed

**Comment by mazbrili** [comment]
@Begasus  please check again.. on my real box its ok..

**Comment by Begasus** [comment]
> please check again

Been busy lately, will check later :)

**Comment by Begasus** [comment]
Deskbar icon is still a problem, no way of closing it beside killing it with PC, aside from that it seems to be working OK.

**Comment by mazbrili** [comment]
> > please check again
> 
> Been busy lately, will check later :)

hehe sorry..

**Comment by mazbrili** [comment]
asking review to @Begasus  and @extrowerk 
thank you very much for both of you

**Comment by Begasus** [comment]
Icon in the Deskbar is closing once you close Elforkane now (it doesn't show any submenu if you right/left click on the icon though)
If you could check that the settings are saved in ~/config/settings/elforkane that would be great :)

**Comment by mazbrili** [comment]
> Icon in the Deskbar is closing once you close Elforkane now (it doesn't show any submenu if you right/left click on the icon though)
> If you could check that the settings are saved in ~/config/settings/elforkane that would be great :)

yeah.. i still mods and test for location issue..

**Comment by mazbrili** [comment]
hope this change fix all issue..  asking review @Begasus and @extrowerk

**Comment by mazbrili** [comment]
thanks all @Begasus  and @extrowerk

**Pull request merged**